### DISCUSSION TITLE
[Feedback requested] Browser access mode for Prometheus, InfluxDB and Elasticsearch

### DISCUSSION THREAD

**Discussion by bijwaard** [grafana/grafana#43019] [Data Sources]
Dear all,

I noticed in the [8.3.0-beta2](https://grafana.com/docs/grafana/latest/release-notes/release-notes-8-3-0-beta2/) release notes, that browser access mode is deprecated for datasources Prometheus, InfluxDB and Elasticsearch. 

I use browser-access mode to use local influxdb databases as datasource that are not remotely accessible by the grafana server, i.e. grafana runs in the cloud and some of my datasources are behind the company firewall. Another use-case I have for browser-access mode is access control by a web proxy that proxies multiple datasources. Access control in the proxy by URL path is much easier to configure and maintain than specifying account+password for every individual datasource.

Some questions:

- Does anybody know why browser access mode is deprecated and in what release it would be removed?
- Are there other use cases for browser access mode?
- Are others also affected by this change?
- What would be necessary to keep browser access mode supported for influxdb and possibly others?

Kind regards,
Dennis

**Comment by bijwaard** [comment]
Hi, I just found an open related ticket  #12757 for elastic search.

**Comment by auwaerter** [comment]
I'm also curious why it is planned to deprecate the browser access mode. In my case, i'm using it for an exposed (but with basic auth protected) prometheus url, where i do have one master grafana and round about 5 prometheus datasources. It is quite easy and convenient to do it like that, but i've never really found a reason for the deprecation. 

What could be the alternative approach when we can not reach the datasource trough the backend?

**Comment by bijwaard** [comment]
Hi Benjamin,
An alternative could be to use server-side datasources with credentials (username, password) in grafana. This would be the way to get alerting to work, and possibly why they want to ditch browser access mode. This may not work with web proxies with authentication/authorization at different levels, I think you can only specify one username+password for a server-side datasource.

For me most datasources are behind an (apache) proxy, the credentials are the same for groups of them (e.g. a certain measurement location). The datasources themselves are not reachable directly, only via the proxy, since they have private IP address and their webserver port is tunneled to the machine that has the proxy. I would rather not want something grafana-specific, since the webserver also serves other applications like web-browsers.

Kind regards,
Dennis

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
